### DISCUSSION TITLE
problem with Test (GTest) integration in Test Explorer

### DISCUSSION THREAD

**Discussion by tofDou** [microsoft/vscode-cmake-tools#3841] [Q&A] [answered]
**Hello everyone,**

With last version of cmake-tools plugin (V1.19.10) and also last rc version of cmake (3.30rc3), **my test explorer and test result tabs remains "empty"** whereas using CTest in CLI works normally and show me different tests in different test suites.
Naturally I have a worse behavior using the last release of both OSS. Since I noticed very recent bug corrections in CMake and vscode plugin dealing with add_test and extra properties ([this commit for instance](https://github.com/Kitware/CMake/commit/a0343abd59dee4d9cf99429dceabc9244cebd446)) and also recent refactoring when discovering tests in vscode plugin, I give it a try...

To be more precise and to give you some inputs:

When I say "empty", the fact is that I have only one item in my tree named as my project name

![image](https://github.com/microsoft/vscode-cmake-tools/assets/82022112/8a8170c5-b2e7-4994-9274-5e78ad9a586d)

**So I search why the test discovery process "fails"**

```bash
ctest --show-only=json-v1 --test-dir /home/tof/WorkSpace/BMS_Project/build/Release/src/tests > /tmp/test_list.json
```
[test_list.json](https://github.com/user-attachments/files/15924637/test_list.json)

Using the old fashion ctest -N is also OK

```bash
ctest -N --test-dir /home/tof/WorkSpace/BMS_Project/build/Release/src/tests
```

returns:
![image](https://github.com/microsoft/vscode-cmake-tools/assets/82022112/d6b183cd-7588-4e7e-afaa-9d6038f4939f)

## My hypothesis:

Problem with CTestArgs during the discovery phase???
In my cas without `--test-dir` option, ctest integration in vscode doesn't work at all since no test where found

**more little things...:**
extract of settings.json:

```json
{
    "cmake.ctestArgs": [
        "--test-dir",
        "/home/tof/WorkSpace/BMS_Project/build/Release/src/tests"
    ],
    "cmake.ctest.testExplorerIntegrationEnabled": true,
    "cmake.ctest.allowParallelJobs": true,
    "cmake.ctest.parallelJobs": 2,
    "cmake.ctestDefaultArgs": [
        "--output-on-failure"
    ]
}
```

When switching `allowParallelJobs` to false , nothings works at all since the ctest command line is completly "broken" with its -R option:

![image](https://github.com/microsoft/vscode-cmake-tools/assets/82022112/15284426-f205-40f9-9612-76fe3ffa58c7)

**Other Note:**
My `CTestTestfile.cmake` is not in the binaryDir but in a subdirectory `build/Release/src/tests/CTestTestfile.cmake`.

![image](https://github.com/microsoft/vscode-cmake-tools/assets/82022112/bbadc07b-da59-46d8-9821-c9850acf0e60)

Same thing for test results

![image](https://github.com/microsoft/vscode-cmake-tools/assets/82022112/94a54a83-2ec3-4aa6-b32b-d7c3526839d2)

## Please help

Any idea to solve this issue for me will be highly appreciated.

**Comment by gcampbell-msft** [comment, marked as answer]
@tofDou Thanks for all of this testing and information! We primarily follow up with an track issues, so if you could update your issue here: #3842 with all of this context, it would make things easier. I'll follow up with some thoughts and questions in the issue.
